File: app/src/main/java/com/sharpdroid/registroelettronico/BottomSheet/AgendaBS.java
package com.sharpdroid.registroelettronico.BottomSheet;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.BottomSheetDialogFragment;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.sharpdroid.registroelettronico.Interfaces.Client.AdvancedEvent;
import com.sharpdroid.registroelettronico.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * <p>A fragment that shows a list of items as a modal bottom sheet.</p>
 * <p>You can show this modal bottom sheet from your activity like this:</p>
 * <pre>
 *     AgendaBS.newInstance(30).show(getSupportFragmentManager(), "dialog");
 * </pre>
 * <p>You activity (or fragment) needs to implement {@link AgendaBS.Listener}.</p>
 */
public class AgendaBS extends BottomSheetDialogFragment {
    private static final int mItemCount = 5;
    private final int texts[] = {
            R.string.condividi_bs,
            R.string.inserisci_calendario_bs,
            R.string.copia_bs,
            R.string.archivia_bs
    };
    private final int icons[] = {
            R.drawable.agenda_bsheet_share,
            R.drawable.agenda_bsheet_calendar,
            R.drawable.agenda_bsheet_copy,
            R.drawable.agenda_bsheet_archive
    };
    private Listener mListener;
    private AdvancedEvent event;

    public void setEvent(AdvancedEvent event) {
        this.event = event;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_item_list_options, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        final RecyclerView recyclerView = (RecyclerView) view;
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(new ItemAdapter(/*getArguments().getInt(ARG_ITEM_COUNT)*/));
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        final Fragment parent = getParentFragment();
        if (parent != null) {
            mListener = (Listener) parent;
        } else {
            mListener = (Listener) context;
        }
    }

    @Override
    public void onDetach() {
        mListener = null;
        super.onDetach();
    }

    public interface Listener {
        void onBottomSheetItemClicked(int position, AdvancedEvent e);
    }

    protected class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.text)
        TextView text;
        @BindView(R.id.layout)
        View layout;
        @BindView(R.id.image)
        ImageView image;

        ViewHolder(View v) {
            super(v);
            ButterKnife.bind(this, v);
            layout.setOnClickListener(v1 -> {
                if (mListener != null) {
                    mListener.onBottomSheetItemClicked(getAdapterPosition(), event);
                    dismiss();
                }
            });
        }
    }

    private class ItemAdapter extends RecyclerView.Adapter<ViewHolder> {
        @Override
        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.fragment_item_agenda_options, parent, false));
        }

        @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            if (position != 0) {
                holder.text.setText(getString(texts[position - 1]));
                holder.text.setTextColor(Color.BLACK);
                holder.image.setImageResource(icons[position - 1]);
            } else {
                holder.text.setText(event.isCompleted() ? "Non completato" : "Completato");
                holder.text.setTextColor(ContextCompat.getColor(getContext(), R.color.intro_blue_dark));
                holder.image.setImageResource(event.isCompleted() ? R.drawable.agenda_uncomplete : R.drawable.agenda_complete);
            }
        }

        @Override
        public int getItemCount() {
            return mItemCount;
        }

    }
}

File: app/src/main/java/com/sharpdroid/registroelettronico/BottomSheet/AgendaBS.kt
package com.sharpdroid.registroelettronico.BottomSheet

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.support.design.widget.BottomSheetDialogFragment
import android.support.v4.content.ContextCompat
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.sharpdroid.registroelettronico.Databases.Entities.SuperAgenda
import com.sharpdroid.registroelettronico.R
import kotlinx.android.synthetic.main.fragment_item_agenda_options.view.*

class AgendaBS : BottomSheetDialogFragment() {
    private val texts = intArrayOf(R.string.condividi_bs, R.string.archivia_bs)
    private val icons = intArrayOf(R.drawable.agenda_bsheet_share, R.drawable.agenda_bsheet_archive)
    private var mListener: Listener? = null
    private var event: SuperAgenda? = null

    fun setEvent(event: SuperAgenda) {
        this.event = event
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.fragment_item_list_options, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        val recyclerView = view as RecyclerView?
        recyclerView?.layoutManager = LinearLayoutManager(context)
        recyclerView?.adapter = ItemAdapter(mListener!!)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        val parent = parentFragment
        if (parent != null) {
            mListener = parent as Listener
        } else {
            mListener = context as Listener?
        }
    }

    override fun onDetach() {
        mListener = null
        super.onDetach()
    }

    interface Listener {
        fun onBottomSheetItemClicked(position: Int, e: SuperAgenda)
    }

    private inner class ItemAdapter(val clickListener: Listener) : RecyclerView.Adapter<ItemAdapter.ViewHolder>() {
        inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
            fun bindData(text: String, color: Int, image: Int, tint: Int?) {
                itemView.text.text = text
                itemView.text.setTextColor(color)
                itemView.image.setImageResource(image)
                itemView.image.setColorFilter(tint ?: ContextCompat.getColor(activity.applicationContext, R.color.icon_on_white))
                itemView.setOnClickListener {
                    clickListener.onBottomSheetItemClicked(adapterPosition, event!!)
                    dismiss()
                }
            }
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.fragment_item_agenda_options, parent, false))
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            when (position) {
                0 -> holder.bindData(if (event!!.completed) "Non completato" else "Completato",
                        ContextCompat.getColor(context, R.color.greenmaterial_800),
                        if (event!!.completed) R.drawable.agenda_uncomplete else R.drawable.agenda_complete, ContextCompat.getColor(context, R.color.greenmaterial_800))
                1 -> holder.bindData(if (event!!.test) "Rimuovi contrassegno" else "Contrassegna come compito",
                        ContextCompat.getColor(context, R.color.redmaterial),
                        if (event!!.test) R.drawable.agenda_uncomplete else R.drawable.agenda_complete, ContextCompat.getColor(context, R.color.redmaterial))
                else -> holder.bindData(getString(texts[position - 2]), Color.BLACK, icons[position - 2], null)

            }
        }

        override fun getItemCount(): Int {
            return mItemCount
        }

    }

    companion object {
        private val mItemCount = 4
    }
}
